### ISSUE TITLE
 Is the library is still being developed? last - on 10 Nov 2014 

### ISSUE THREAD

**Issue by kamilwlf** [roboguice/roboguice#327]
trying to choose the best library for DI? Is the library is still being developed? 

my understanding:
Butterknife - simple and works without restrictions,only basic functions
Dagger- make problems solvable by using code generation, hand written code, as if we were writing all the code that creates and provides our dependencies ourselves. It requires a lot of effort
AndroidAnnotations - replaces both Butterknife and  Dagger. Usage is reasonably easy to. A massive (in my mind) drawback is the generation for sub-classes of EVERY class you annotate. This quickly becomes a bit messy and is not the nicest way to go about things

Is it worth entering in roboguice?

**Comment by OlegDokuka** [comment]
+1

**Comment by mdelolmo** [comment]
No one has even replied? I guess that's a bad sign...

**Comment by nicolasjafelle** [comment]
Better move to guice, is still active and does not depends on android support libraries to keep up to date... Just inject/provide the context.

**Comment by stephanenicolas** [comment]
@Nicolas Jafelle, RoboGuice is much better than Guice for Android, the
performances of RG, when using RoboBlender are far better than what Guice
can give.

We are currently trying our best to improve RG still, but we are facing
limits inherent to Guice. Guice could have been coded in a different way
and provide a much better runtime performance, but it's not the case.
Hence, our efforts to get even more performance out of RG are facing this
limit.

But still, we plan to release a new version of RG. We don't know exactly
when.

S.

2015-12-09 22:51 GMT-05:00 Nicolas Jafelle <notifications@github.com>:

**Comment by nicolasjafelle** [comment]
@stephanenicolas thanks for your quick response and the update! Do you think that dagger will be The DI library?

**Comment by stephanenicolas** [comment]
Dagger is pretty efficient, RoboGuice is actually very close in terms of
performance. Dagger 2 goes faster but requires more boiler plate.
RG requires less configuration and is more advanced. You got everything to
decide which one to use :)

2015-12-10 10:16 GMT-05:00 Nicolas Jafelle <notifications@github.com>:

**Comment by nicolasjafelle** [comment]
Well... I have been using Roboguice since 1.2 or something like that, dagger for me is still to much boiler plate code and combined with Rx or Otto or another libraries make dagger as you said to much boiler plate code by providing all the injections and that. Thanks for all your help!

**Comment by erikpragt-siteminder** [comment]
Maybe this is a nice read for this thread? http://blog.nimbledroid.com/2016/03/07/performance-of-dependency-injection-libraries.html

**Comment by stephanenicolas** [comment]
Yes, it is an interesting thread to read. As noted in the comment section,
we are not sure about the numbers mentioned in the article and would have
loved to see other aspects of the frameworks evaluated but that's the way
it is.

We have released RG 4 on central, we got huge improvements with that
release buy we didn't find time yet to finalize the release, announce it,
write migration docs and so on. Hopefully we can do that soon,
S.

**Comment by nicolasjafelle** [comment]
@stephanenicolas that is true you release version 4.0 but without any news, post, readme update, why? Does it really improves only performance and nothing more?
